src/utils/vc/crypto/Bls12381BBSKeyPairDock2023.js

```diff
@@ -0,0 +1,52 @@
+import {
+  BBSKeypair,
+  BBSSignature,
+  BBSPublicKey,
+  BBSSecretKey,
+  BBSSignatureParams,
+} from '@docknetwork/crypto-wasm-ts';
+
+import { BBS_SIGNATURE_PARAMS_LABEL_BYTES } from '@docknetwork/crypto-wasm-ts/lib/anonymous-credentials';
+
+import { Bls12381BBS23DockVerKeyName } from './constants';
+import DockCryptoKeyPair from './common/DockCryptoKeyPair';
+
+export default class Bls12381BBSKeyPairDock2023 extends DockCryptoKeyPair {
+  constructor(options) {
+    super(options, Bls12381BBS23DockVerKeyName);
+  }
+
+  static generate({
+    seed, params, controller, id, msgCount = 1,
+  } = {}) {
+    const keypair = BBSKeypair.generate(
+      params
+        || BBSSignatureParams.getSigParamsOfRequiredSize(
+          msgCount,
+          BBS_SIGNATURE_PARAMS_LABEL_BYTES,
+        ),
+      seed,
+    );
+    return new Bls12381BBSKeyPairDock2023({ keypair, controller, id });
+  }
+
+  static signerFactory(key) {
```

Comment:
For clarify, rename `signerFactory` of parent class as its signature is different from this.